public/components/custom_panels/helpers/utils.tsx

```diff
@@ -397,14 +400,11 @@ export const renderCatalogVisualization = async ({
   const catalogSourceName = catalogSource.split('.')[0];
   const catalogTableName = catalogSource.split('.')[1];
 
-  const defaultAggregation = 'avg'; // pass in attributes to this function
-  const attributes: string[] = [];
-
   const visualizationQuery = updateCatalogVisualizationQuery({
     catalogSourceName,
     catalogTableName,
-    aggregation: defaultAggregation,
-    attributesGroupBy: attributes,
+    aggregation: queryMetaData!.aggregation,
+    attributesGroupBy: queryMetaData!.attributesGroupBy,
```

Comment:
why does it first declare `queryMetaData?: MetricType['query'];` as nullable, then use the non-null assertions?

Reply:
Corrected.